src/cargo/sources/registry/remote.rs

```diff
@@ -21,29 +23,76 @@ use std::task::{ready, Poll};
 /// A remote registry is a registry that lives at a remote URL (such as
 /// crates.io). The git index is cloned locally, and `.crate` files are
 /// downloaded as needed and cached locally.
+///
+/// This type is primarily accessed through the [`RegistryData`] trait.
+///
+/// Using Git to host this index provides some benefits. The most prominent one
+/// is that the entire index can be stored efficiently locally on disk. This
+/// means that all queries of a registry can happen locally and don't need to
+/// touch the network.
+///
+/// See the [module-level documentation](super) for the index format and layout.
+///
+/// ## Git-based index was the default before 1.70.0.
```

Comment:
Trade offs being discussed in two different sections feels weird. Depending on how someone skims this, they will likely only see one or the other.